src/RequirementChecker/AppRequirementsFactory.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the box project.
+ *
+ * (c) Kevin Herrera <kevin@herrera.io>
+ *     Théo Fidry <theo.fidry@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace KevinGH\Box\RequirementChecker;
+
+use Assert\Assertion;
+use KevinGH\Box\Json\Json;
+use UnexpectedValueException;
+use function array_diff_key;
+use function array_key_exists;
+use function array_map;
+use function sprintf;
+use function str_replace;
+use function substr;
+
+/**
+ * Collect the list of requirements for running the application.
+ *
+ * @private
+ */
+final class AppRequirementsFactory
+{
+    private const SELF_PACKAGE = '__APPLICATION__';
+
+    /**
+     * @param string $composerLockContents JSON contents of the `composer.lock` file
+     *
+     * @return array Serialized configured requirements
+     */
+    public static function create(string $composerLockContents): array
+    {
+        $lockDecodedContents = (new Json())->decode($composerLockContents, true);
```

Comment:
since the content is decoded in several places, maybe this should be done in `Configuration` in a dedicated PR